bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java

```diff
@@ -225,8 +225,9 @@ public void schedule(final Runnable task, final Expression expression) {
 
         @Override
         public boolean remove(Runnable task) {
-            if (futures.get(task) != null && futures.get(task).cancel(false)) {
-                running.remove(task);
+            Future<?> future = futures.get(task);
+            if (future != null && future.cancel(false)) {
+                running.remove(future);
```

Comment:
ScheduledFutureTask is implementing both the Runnable and Future interfaces. Could be the source of your confusion? Also, beware, schedule() is overridden. You refer to https://github.com/kaikreuzer/smarthome/blob/7d7ef907d709afe015a0c035b60bd2602307895f/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java#L212 , but then the other implementation is called at https://github.com/kaikreuzer/smarthome/blob/7d7ef907d709afe015a0c035b60bd2602307895f/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java#L162